pkg/e2etemplates/render.go

```diff
@@ -0,0 +1,34 @@
+package e2etemplates
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/giantswarm/microerror"
+)
+
+type Data interface {
+	// Validate returns error matched by IsInvalidData when template data
+	// is invalid.
+	Validate() error
+}
+
```

Comment:
Example usage: `Render(AWSOperatorChartValues, AWSOperatorChart{})`. Ofc this call would fail with `invalidDataError`.